src/coreclr/vm/callstubgenerator.cpp

```diff
@@ -1677,10 +1691,20 @@ void CallStubGenerator::ProcessArgument(ArgIterator *pArgIt, ArgLocDesc& argLocD
     // we always process single argument passed by reference using single routine.
     if (pArgIt != NULL && pArgIt->IsArgPassedByRef())
     {
-        _ASSERTE(argLocDesc.m_cGenReg == 1);
-        pRoutines[m_routineIndex++] = GetGPRegRefRoutine(argLocDesc.m_idxGenReg);
-        pRoutines[m_routineIndex++] = pArgIt->GetArgSize();
-        m_r1 = NoRange;
+        if (argLocDesc.m_cGenReg == 1)
+        {
+            pRoutines[m_routineIndex++] = GetGPRegRefRoutine(argLocDesc.m_idxGenReg);
+            pRoutines[m_routineIndex++] = pArgIt->GetArgSize();
+            m_r1 = NoRange;
+        }
+        else
+        {
+            _ASSERTE(argLocDesc.m_byteStackIndex != -1);
+            pRoutines[m_routineIndex++] = GetStackRefRoutine();
+            pRoutines[m_routineIndex++] = ((int64_t)pArgIt->GetArgSize() << 32) | argLocDesc.m_byteStackIndex;
```

Comment:
Is there a comment somewhere explaining this bit packing?

Reply:
No, it is just using 4 bytes for arg size and 4 bytes for stack offset to not to waste space unnecessarily.